src/lnp/application/message.rs

```diff
@@ -591,6 +634,237 @@ pub struct ChannelReestablish {
+    pub len: u16,
+
+    /// feature bytes
+    pub features: Vec<u8>,
+
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// Short channel ID
+    pub short_channel_id: ShortChannelId,
+
+    /// Node Id 1
+    pub node_id_1: PublicKey,
+
+    /// Node Id 2
+    pub node_id_2: PublicKey,
+
+    /// Bitcoin key 1
+    pub bitcoin_key_1: PublicKey,
+
+    /// Bitcoin key 2
+    pub bitcoin_key_2: PublicKey,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct NodeAnnouncements {
+    /// Signature
+    pub signature: Signature,
+
+    /// length of feature bytes
+    pub flen: u16,
+
+    /// feature bytes
+    pub features: Vec<u8>,
+
+    /// time stapm
+    pub timestamp: u32,
+
+    /// Node Id
+    pub node_id: PublicKey,
+
+    /// RGB colour code
+    pub rgb_color: NodeColor,
+
+    /// Node Alias
+    pub alias: Alias,
+
+    /// length of node address bytes
+    pub addrlen: u16,
+
+    /// Node address
+    pub address: Vec<u8>,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct ChannelUpdate {
+    /// Signature
+    pub signature: Signature,
+
+    /// Chainhash
+    pub chain_hash: AssetId,
+
+    /// Short Channel Id
+    pub short_channel_id: ShortChannelId,
+
+    /// Time stamp
+    pub timestamp: u32,
+
+    /// message flags
+    pub message_flags: u8,
+
+    /// channel flags
+    pub channle_flags: u8,
+
+    /// cltv expiry delta
+    pub cltv_expiry_delta: u16,
+
+    /// minimum HTLC in msat
+    pub htlc_minimum_msal: u64,
+
+    /// base fee in msat
+    pub fee_base_msat: u32,
+
+    /// fee proportional millionth
+    pub fee_proportional_millionths: u32,
+
+    /// if option_channel_htlc_max is set
+    pub htlc_maximum_msat: u64,
+}
+
+/// Extended Gossip messages
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct QueryShortChannelIds {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// short ids to query
+    pub short_ids: Vec<ShortChannelId>,
+
+    /// short id tlv stream
+    #[tlv(type = 1)]
+    pub short_id_tlvs: BTreeMap<u8, Vec<u8>>, /* is this the correct way to
+                                               * write tlv streams? */
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct ReplyShortChannelIdsEnd {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// full information
+    pub full_information: u8,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, Debug, Display, LightningEncode, LightningDecode,
+)]
+#[lnpbp_crate(crate)]
+#[display(Debug)]
+pub struct QueryChannelRange {
+    /// chain hash
+    pub chain_hash: AssetId,
+
+    /// first block number
+    pub first_block_num: u32,
```

Comment:
Let's match the name from the spec:
```suggestion
    pub first_blocknum: u32,
```

Reply:
Done